lib/fs.js

```diff
@@ -1469,16 +1481,122 @@ fs.unwatchFile = function(filename, listener) {
   }
 };
 
-// Regexp that finds the next partion of a (partial) path
-// result is [base_with_slash, base], e.g. ['somedir/', 'somedir']
-const nextPartRe = isWindows ?
-  /(.*?)(?:[\/\\]+|$)/g :
-  /(.*?)(?:[\/]+|$)/g;
+// Functions that find the next portion of a (partial) path up to and including
+// a possible path separator (e.g. 'foo' or 'foo/')
+function findNextPartWin32(path, start) {
+  var i = start;
+  for (; i < path.length; ++i) {
+    const code = path.charCodeAt(i);
+    if (code === 47/*/*/ || code === 92/*\*/)
```

Comment:
can you quote and space these so it doesn't just look like weird comments? 
`if (code === 47 /* '/' */ || code === 92 /* '\' */)`
or
`if (code === 47 || code === 92) // '/' or '\'`
I personally find the latter clearest